code/game/turfs/turf.dm

```diff
@@ -184,8 +184,10 @@
 		return TRUE
 	if(can_lay_cable() && istype(C, /obj/item/stack/cable_coil))
 		var/obj/item/stack/cable_coil/coil = C
-		for(var/obj/structure/cable/LC in src)
-			return
+		var/obj/structure/cable/found = locate(/obj/structure/cable) in src
```

Comment:
You should check all cables in the lawn. Not only the first one found.
This work for one existed cable in turf, but what if we place more then 2 in turf?
I think there need something like 
`for(var/thing in contents)
	var/obj/structure/cable/found = thing
	if(found)
		if(found.color == C.color)
			return FALSE`

Reply:
Actually
```
for(var/obj/structure/cable/found in contents)
		if(found && found.color == C.color)
			return FALSE
```
This loop needs to be typed.